### ISSUE TITLE
findbugs complains about dead assignments

### ISSUE THREAD

**Issue by haroldcarr** [immutables/immutables#373]
complains about the final assignment to `bits` in the generated method:

    private void from(Object object) {
      long bits = 0;
      if (object instanceof SearchableEntity) { 
          ...
      }
      if (object instanceof Identifiable) {
        ....
      }
      if (object instanceof AbstractSearchableApiCollectionProvenance) {
        AbstractSearchableApiCollectionProvenance instance = (AbstractSearchableApiCollectionProvenance) object;
        withNamespace(instance.getNamespace());
        if ((bits & 0x1L) == 0) {
          withName(instance.getName());
          bits |= 0x1L;
        }
        withProvenance(instance.getProvenance());
        if ((bits & 0x2L) == 0) {
          withId(instance.getId());
          bits |= 0x2L;
        }
        withVersion(instance.getVersion());
      }
    }

**Comment by elucash** [comment]
I don't know if it's worth "fixing", but I'll see what we can do about this. Although it is not clear why `@SuppressFBWarnings` had no effect?

**Comment by haroldcarr** [comment]
I agree it is low priority.  I have an entry in my FB exclude file to ignore it for now.

Note: looking at the generated code is see `@SuppressWarnings("all")`   but I do NOT see `@SuppressFBWarnings`.

**Comment by elucash** [comment]
we probably generate it only if we have such annotation in the classpath, it might not account for many scenarios with tools/plugins that are actually used

**Comment by elucash** [comment]
I found it's quite hard to keep track of whether write was last during generation without unnecessarily complicating the template. Another approach can be to insert assertion on something like `assert Long.bitCount(bits) >= 0;` but this is also too ugly. Currently, I would suggest to add `edu.umd.cs.findbugs.annotations.SuppressFBWarnings` to the classpath so generated code would be generated with this annotation and warnings suppressed.

**Comment by haroldcarr** [comment]
Understood.  For now I have done:

    <FindBugsFilter>
    
         <Match>
            <Or>
                <Class name="com.example.MyClass$Builder" />
                ...
            </Or>
            <Method name="from"/>
            <Bug pattern="DLS_DEAD_LOCAL_STORE"/>
        </Match>
        ...
